packages/jest-cli/src/watch.js

```diff
@@ -86,7 +86,9 @@ const watch = (
       results => {
         isRunning = false;
         hasSnapshotFailure = !!results.snapshot.failure;
-        testWatcher.setState({interrupted: false});
+        if (config.bail) {
```

Comment:
I think the test interruption issue was introduced here #2312 and it is happening in master...

@cpojer  Is this the right fix? I think that we should only set `{interrupted: false}` when using `bail`... I decided to fix it as part of this PR but it's fine if you prefer me to do it as a separate one that just fixes the test interruption bug.